react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js

```diff
@@ -8,16 +8,26 @@ const initialState = {
 
 function reducer(state = initialState, action) {
   switch (action.type) {
-    case actions.GET_BOOKS: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_TO_CART: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.REMOVE_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.SEARCH_ITEM: // TODO to implement the logic
-      return { ...state };
+    case actions.GET_BOOKS:
+      return { ...state, books: action.payload, originalData: action.payload };
+    case actions.ADD_TO_CART:
+      return { ...state, bookSelected: [...state.bookSelected, action.payload] };
+    case actions.ADD_ITEM: {
+      const books = [...state.bookSelected];
+      const book = books.find(({ id }) => id === action.payload);
```

Comment:
We usually define the logic behind the *how* to update the state to the _actions_. 
In the reducer, we usually simply receive the new state and update it.
In this case, what would seem logical to me is that the payload contains the book to update and this reducer should only do something like:
```
return {
  ...state,
  bookSelected: [...state.bookSelected, action.payload]
}
```